### DISCUSSION TITLE
Full log of copilot activity

### DISCUSSION THREAD

**Discussion by jerometremblay** [aws/copilot-cli#5487] [General]
When we run copilot, the interface is very dynamic and pretty to look at.
Is there a way to also generate a complete text-mode log instead of the dynamic view?

**Comment by Lou1415926** [comment]
@jerometremblay Thank you for the nice words!

Unfortunately not today : ( Are you looking to store the logs for future references?

**Reply by jerometremblay** [reply]
Yes, and also to more easily let a task run in the background and see what happened step by step. Otherwise, when I come back I just see "everthing rolled back" and i don't know exactly what worked and what did not.

**Reply by Lou1415926** [reply]
Gotcha. Thanks for sharing 👍🏼 !

 Copilot should not have attempted to erase the stack events that the deployment ended with, which should contain information of what failed and why. Did you see something like ⬇️, or did you see only the last line "deploy service: stack xxx did not complete successfully", or was the deployment progress tracker you were left with too scrambled that it became unreadably?
<img width="900" alt="Screenshot 2023-11-17 at 1 18 06 PM" src="https://github.com/aws/copilot-cli/assets/79273084/f5c452cb-a20f-4202-9f7e-2100e60c3c15">

To clarify, I still agree that having a text based output to certain log files can be a helpful addition. I am trying to identify whether or not it was because of a bug that you weren't able to see the failure reasons.

**Reply by jerometremblay** [reply]
Searched but couldn't find the exact example I had in mind for this. I did remember that it was as you showed in your screenshot. A bunch of resources were all in a rolled back state. It's just that since the execution order is not fixed in this view, i wasn't sure which one exactly failed and triggered the rollback. 

Since then I fould the `--no-rollback` option that helps a lot for this particular case.

**Comment by jerometremblay** [comment]
Example of lost information

<img width="669" alt="Capture d’écran, le 2023-12-03 à 13 16 17" src="https://github.com/aws/copilot-cli/assets/4490798/02282d51-bb48-48e9-8b11-bcba9a8c433f">

But after everything is done, i have this
<img width="1295" alt="Capture d’écran, le 2023-12-03 à 13 15 44" src="https://github.com/aws/copilot-cli/assets/4490798/e92533db-946c-46c6-b596-646cf0152e12">

So I don't know that the problem is caused by something in Route 53. (An extra CNAME in the zone so it cannot be deleted)

**Reply by Lou1415926** [reply]
Got it..! I think you are right - for delete commands, the progress are indeed erased after they are done: https://github.com/aws/copilot-cli/blob/90b23b0a59b36d035e3d86e4345bb15890960fa9/internal/pkg/deploy/cloudformation/cloudformation.go#L800. It's a good idea to preserve them for troubleshooting purposes 👍🏼
